### PR TITLE
Adding support for Minecraft 1.6

### PR DESCRIPTION
**Pull request by shartte**
This pull request should add basic support for Minecraft 1.6

It does the following:
- Adds support to parse the Minecraft JSON blobs that specify library dependencies (with tests)
- Parses the JSON included with FML to add build dependencies and download natives to jars/bin/natives
- Fixes fixing the *.exc file (Contained some invalid java signatures in 1.6, it now skips processing those and writes them to the output mapping unchanged)
- Downloads the Minecraft client using the new URL format (URLs are no longer specified in FML config)

Current issues I know of:
- When switching MC versions, one has to do a manual removal of the build directory (clean doesnt seem to pick it up), since the extracted deobfuscated jar is still there and gets picked up by the decompiler. This causes compilation issues because MinecraftApplet and other classes are gone in 1.6
- Didn't test Eclipse integration (since i don't have it installed).

This should be considered a starting point however :-)

### CODE REVIEW

**Comment by AbrarSyed** [comment]
@shartte get on our IRC, the #forgeessentials one, or the #minecraftforge one on esper.net

We can talk easier there.

I must say, good work... but since I didnt push the changes i had already made, I can't exactly pull this as it is. Don't worry, I will definitely steal some of your code as I complete the update.

**Comment by AbrarSyed** [comment]
the issues you specified

1) I a havent tested this thoroughly, but my use of gradle's CI support should allow it to detect that the downloaded version of forge is out of date, and thus redo everything related to minecraft including decompiling it again.

2) eclipse integration should be fine, provided you used the gmcp dependency configuration for the 1.6 stuff as I did for the 1.5.

**Comment by shartte** [comment]
I'll see that I hop on IRC once I am back from work.

I wasn't sure about the intended level of Java compatibility, so I went for named RegExp groups (the RegExp is quite hard to parse otherwise). Will fix that in the pull request.
